core/datasource/common/time.go

```diff
@@ -115,11 +127,22 @@ func InternalTimeTriggerFromProto(
 		Every:   protoTrigger.Every,
 	}
 
-	tt.SetNextTrigger(timeNow)
+	if initial != nil {
+		tt.SetNextTrigger(timeNow)
+	}
+
 	return tt
 }
 
-type InternalTimeTriggers [1]InternalTimeTrigger
+type InternalTimeTriggers [1]*InternalTimeTrigger
+
+func (i InternalTimeTriggers) Empty() error {
```

Comment:
This is a bit weird, I would expect `T.Empty()` to return a bool, not an error if `Empty() == true`

Reply:
yea, but I wanted to return the error down from the type for some reason. but agree it's weird.